### DISCUSSION TITLE
Avalonia in 2024 - Balancing Growth and Sustainability

### DISCUSSION THREAD

**Discussion by MikeCodesDotNET** [AvaloniaUI/Avalonia#17661] [General]
It's that time of year again when we share how we're working to make Avalonia a sustainable OSS project. 

This year, we've shared more data and more context around our decisions, aiming to provide a clearer picture of where we are and where we're heading. We believe this transparency helps foster better discussions about Avalonia's future, and look forward to hearing your thoughts. 

**[Read the post here](https://avaloniaui.net/blog/avalonia-ui-in-2024-growth-challenges-and-the-road-ahead)**

**Comment by robloo** [comment]
Great writeup as usual!

> One experiment involved offering lower-cost support agreements to make our offerings more accessible to independent developers and small companies. This was inspired by community feedback, with many citing pricing as a barrier to adoption.
>
> The results were enlightening. Despite vocal community requests for lower prices, our indie-focused tiers accounted for just 0.625% of our annual revenue. This "say-do gap", where potential customers advocate strongly for lower prices but don't convert when those prices are offered, is a well-documented phenomenon that we've now experienced firsthand

As a strong advocate for lower-cost pricing I'm still not sure the reasons are fully understood. This is best termed as a "loss-leader". It's a product that doesn't make money but helps draw people in and improve brand and public sentiment. The times it's really justifiable to cut an offering like this is when it is cannibalizing higher-tier sales or taking too many resources from other income streams.

On the resource usage front that may indeed be a problem. As with all support-type offerings team member time is at a premium. Perhaps it could have been limited in more ways (N-number of hours per Month). On cannibalizing higher-tier sales: you have indicated in the past companies would not want to pay for more expensive contracts. However, I believe the indie plan was -- or could have been -- limited quite a bit on size of company (dev size, gross income).

In the back of my mind, the goal of this type of product is also to get in early on projects and grow with them. A very small percentage will be successful long term but you still want to be with them from the beginning if possible. That said, a support agreement certainly isn't stopping personal use of Avalonia by devs who can then advocate and push for it within their organization over time (an extremely important entry point). 

At the end of the day hopefully Avalonia Accelerate is a better match for this type of revenue stream and vanguard product anyway.

**Reply by MikeCodesDotNET** [reply]
Thanks @robloo. 

We attempted to make the Indie support tier as limited as possible. For example, we limited the number of tickets organisations could submit, but ultimately, we needed to sell more agreements to warrant its continued existence as it impacted enterprise sales. 

The straw that broke the camel's back was during a conversation with a large enterprise team developing with Avalonia. Despite their $700 million annual revenue, they sought a single Indie agreement for their entire development team. When I suggested that the enterprise tier would be more appropriate, they responded, "We're not spending that much on an OSS project."

This sentiment likely echoed what many enterprises felt but didn't openly say: the Indie support offering inadvertently devalued the enterprise tier. Enterprises perceived the situation as being penalised for their size, while we viewed it as offering smaller developers a fair discount. Moving forward, we will happily discuss discounts to smaller teams privately for those with limited financial resources who want to work with us. 

That said, I think the community is in a great position regarding support. GitHub Discussions and Telegram conversations are lively, enabling most developers to find help without paid support. Paid tiers cater primarily to organisations that need SLAs and want to contribute directly to Avalonia's sustainability. It's not really an indie-centric offering anymore.

Avalonia Accelerate, however, is designed with indie developers in mind. We've taken previous feedback seriously and are structuring Accelerate licensing and pricing to appeal to individuals.

**Reply by robloo** [reply]
Thanks for all the great details. I can certainly understand and appreciate the decision here.

> Enterprises perceived the situation as being penalised for their size, while we viewed it as offering smaller developers a fair discount. Moving forward, we will happily discuss discounts to smaller teams privately for those with limited financial resources who want to work with us.

This is a great insight on differing viewpoints. The last comment also seems like a good compromise for those needing smaller or more tailored support agreements. It looks like there is still a way to meet everyone's needs!

> Avalonia Accelerate, however, is designed with indie developers in mind. We've taken previous feedback seriously and are structuring Accelerate licensing and pricing to appeal to individuals.

I look forward to its release!

**Comment by JakeSays** [comment]
That $700MM enterprise customer mentioned is the kind of customer that is causing a lot of pain in OSS.
